src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
We could probably even simply the implementation a lot. Something like the following should probably do the trick.

```
function processGroup(group) {
  var items = [{
    id: group.id,
    isGroup: true
  }];
  
  return Array.prototype.concat.apply(items,
    group.apps.map(processApp).concat(group.groups.map(processGroup))
  );
}
```
*Sketchy implementation, use at your own risk. :)*